src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,6 +32,25 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
+  updateStatus(id: string, status: string): Observable<void> {
+    const tasks = this.readTasks();
+    const updated_tasks = tasks.map(task => {
+      if (task.id === id) {
+        task.status = status;
+      }
+      return task;
+    });
+    this.writeTasks(updated_tasks);
+    return of(null);
+  }
+
+  clearCompletedTasks(): Observable<void> {
+    const tasks = this.readTasks();
+    const clearedTasks = tasks.filter(task => task.status !== 'done');
```

Comment:
to make it a bit cleaner, one could have done using or defining some sort of types/typing instead of strings

Reply:
Updated with property check